### PR TITLE
correctly cancel repeat records if repeater missing

### PR DESCRIPTION
**Pull request by snopoke**
Introduced in d54cd3dccd912d767c43b72381c1468f255108c0

### CODE REVIEW

File: corehq/motech/repeaters/tasks.py [hunk 1]
```diff
@@ -71,19 +71,18 @@ def process_repeat_record(repeat_record):
...
 
     try:
-        if repeat_record.repeater and repeat_record.repeater.paused:
+        if repeater and repeater.paused:
             # postpone repeat record by 1 hour so that these don't get picked in each cycle and
             # thus clogging the queue with repeat records with paused repeater
             repeat_record.postpone_by(timedelta(hours=1))
             return
-        if repeat_record.repeater.doc_type.endswith(DELETED_SUFFIX):
+        if repeater.doc_type.endswith(DELETED_SUFFIX):
```
Comment (mkangia) [inline corehq/motech/repeaters/tasks.py]: should this check for `repeater` as well?

**Review by mkangia — APPROVED** [review]
Thanks.

File: corehq/motech/repeaters/tasks.py (hunk 1 above)
Comment (kaapstorm) [reply corehq/motech/repeaters/tasks.py]: You could `return` after `repeat_record.save()` on line 77. Then you could drop the `repeater and` check on line 80.

Comment (snopoke) [reply corehq/motech/repeaters/tasks.py]: :+1: 519df5a0b6e

**Comment by mkangia** [comment]
sorry, didn't notice that build still needs to run. Will check back on it.
https://travis-ci.org/dimagi/commcare-hq/builds/495882035?utm_source=github_status&utm_medium=notification

**Comment by mkangia** [comment]
sorry about that. Just realised that the build would not run since the branch is deleted.
I guess it was a safe change so this is fine.

**Pull request merged**